packages/esbuild-plugin/src/index.ts

```diff
@@ -115,6 +115,88 @@ function esbuildDebugIdInjectionPlugin(): UnpluginOptions {
   };
 }
 
+function esbuildModuleMetadataInjectionPlugin(injectionCode: string): UnpluginOptions {
+  const pluginName = "sentry-esbuild-module-metadata-injection-plugin";
+  const stubNamespace = "sentry-module-metadata-stub";
+
+  return {
+    name: pluginName,
+
+    esbuild: {
+      setup({ onLoad, onResolve }) {
+        onResolve({ filter: /.*/ }, (args) => {
+          if (args.kind !== "entry-point") {
+            return;
+          } else {
+            return {
+              pluginName,
+              // needs to be an abs path, otherwise esbuild will complain
+              path: path.isAbsolute(args.path) ? args.path : path.join(args.resolveDir, args.path),
+              pluginData: {
+                isProxyResolver: true,
```

Comment:
I am thinking that we should probably chose a slightly different name for the flag here so we avoid a collision with the debug ID injection plugin.

Same goes for the `sentryProxyModule` suffix.